lib/modules.nix

```diff
@@ -36,6 +36,28 @@ in
         helpers = self;
       } // extraSpecialArgs;
     };
+
+  /**
+    Apply an `extraConfig` definition, which should produce a `config` definition.
+    As used by `mkVimPlugin` and `mkNeovimPlugin`.
+
+    The result will be wrapped using a `mkIf` definition.
+  */
+  applyExtraConfig =
+    {
+      extraConfig,
+      cfg,
+      opts,
+      enabled ? cfg.enable or throw "`enabled` argument not provided and no `cfg.enable` option found.",
+    }:
```

Comment:
Thoughts on naming and how generic this should be?

E.g. we could refer to a `varArgFn` (or simply `fn` or `function`) instead of an `extraConfig`

Perhaps this should be broken down into two pieces:
- a generic function living in `utils` that can apply a variable number of arguments to a fn
- a extraConfig-specific helper that applies varargs and then wraps the result in `mkIf`

A generic apply helper might look like:

<details><summary>Details</summary>
<p>


```nix
# While fn is a function, apply aruments from args to it
# Once fn is no-longer a function, its value is retuned
applyArgsIfFn =
  args: fn:
  lib.pipe fn (map (arg: f: if isFunction f then f arg else f) args);
```

```nix
# Using recursion _may_ be more efficient,
# although `lib.drop` looks more expensive than you'd expect...
applyArgsIfFn =
  args: fn:
  if isFunction fn then
    applyArgsIfFn (drop 1 args) (fn (head args))
  else
    fn;
```

```nix
# This recursive impl avoids needing `lib.drop`,
# by doing the recursion on an index instead of the list itself
applyArgsIfFn =
  args: fn:
  let
    len = length args;
    go =
      index: acc:
      if index < len && isFunction acc then
        # If acc is still a fn, pass it args[i], then go a step deeper...
        go (index + 1) (acc (elemAt args index))
      else
        acc;
  in
    go 0 fn;
```

</p>
</details>

Reply:
I can't think of anything more generic for the current implementation since it's specific to extra config and its arguments. In the future, if we find use for the function and want to break it up it could make sense to go that generic route.